### PR TITLE
optimize strava to garmin script

### PR DESCRIPTION
**Pull request by superleeyom**

### CODE REVIEW

File: scripts/garmin_sync.py [hunk 1]
```diff
@@ -214,6 +217,34 @@ async def upload_activities(self, files):
...
+    async def upload_activities_original(self, datas):
+        print("start upload activities to garmin!!!")
+        if not self.is_login:
+            self.login()
+        for data in datas:
+            print(data.filename)
+            with open(data.filename, "wb") as f:
+                for chunk in data.content:
+                    f.write(chunk)
+            f = open(data.filename, "rb")
```
Comment (yihong0618) [inline scripts/garmin_sync.py:229]: f not close.

**Review by yihong0618 — COMMENTED** [review]
will check later, maybe we do not need the stravaWeb

File: scripts/strava_to_garmin_sync.py [hunk 2]
```diff
@@ -73,31 +73,34 @@ def make_gpx_from_points(title, points_dict_list):
...
-        filters = {}
+        print("no activity")
+        return files_list
     else:
         # is this startTimeGMT must have ?
         after_datetime_str = last_activity[0]["startTimeGMT"]
         after_datetime = datetime.strptime(after_datetime_str, "%Y-%m-%d %H:%M:%S")
+        after_datetime = (
+            after_datetime.replace(tzinfo=timezone.utc)
+                .astimezone(tz=pytz.timezone("Asia/Shanghai"))
```
[the remark this answers is not in the document]
Comment (superleeyom) [reply scripts/strava_to_garmin_sync.py:88]: ok，I'll remove it

File: scripts/garmin_sync.py (hunk 1 above)
Comment (superleeyom) [reply scripts/garmin_sync.py:229]: ok,thanks

File: scripts/strava_to_garmin_sync.py [hunk 3]
```diff
@@ -73,31 +72,33 @@ def make_gpx_from_points(title, points_dict_list):
...
-            file = BytesIO(bytes(gpx_doc, "utf8"))
-            files_list.append((file, garmin_type))
-    await garmin_client.upload_activities(files_list)
+    for i in strava_activities[:len(strava_activities)]:
+        try:
+            print(i.id)
+            data = strava_web_client.get_activity_data(i.id, fmt=format)
+            files_list.append(data)
+        except Exception as ex:
+            print(ex)
```
Comment (yihong0618) [inline scripts/strava_to_garmin_sync.py:100]: add some info here

File: scripts/strava_to_garmin_sync.py [hunk 4]
```diff
@@ -73,31 +72,33 @@ def make_gpx_from_points(title, points_dict_list):
...
-        points = generate_strava_run_points(start_time, s)
-        if points:
-            garmin_type = STRAVA_GARMIN_TYPE_DICT.get(activity_type, "running")
-            gpx_doc = make_gpx_from_points("test", points)
-            file = BytesIO(bytes(gpx_doc, "utf8"))
-            files_list.append((file, garmin_type))
-    await garmin_client.upload_activities(files_list)
+    for i in strava_activities[:len(strava_activities)]:
+        try:
+            print(i.id)
```
Comment (yihong0618) [inline scripts/strava_to_garmin_sync.py:96]: drop this print or add some info maybe better

**Review by yihong0618 — COMMENTED** [review]
And you can delete my dead code in this sync file

File: scripts/strava_to_garmin_sync.py (hunk 4 above)
Comment (superleeyom) [reply scripts/strava_to_garmin_sync.py:96]: ok.Let me change that

File: scripts/strava_to_garmin_sync.py (hunk 3 above)
Comment (superleeyom) [reply scripts/strava_to_garmin_sync.py]: @yihong0618 done

**Pull request merged**